### PR TITLE
tile oneapi port

### PR DESCRIPTION
**Pull request by pv-pterab-s**
tile oneapi port. passes up to jit. double not supported

Checklist
---------
- [X] Rebased on latest master
- [X] Code compiles
- [ ] Tests pass
- [ ] Functions added to unified API
- [ ] Functions documented

### CODE REVIEW

File: src/backend/oneapi/kernel/tile.hpp [hunk 1]
```diff
@@ -0,0 +1,110 @@
...
+    auto local = sycl::range(TX, TY);
+
+    int blocksPerMatX = divup(out.info.dims[0], TILEX);
+    int blocksPerMatY = divup(out.info.dims[1], TILEY);
+    auto global       = sycl::range(local[0] * blocksPerMatX * out.info.dims[2],
+                                    local[1] * blocksPerMatY * out.info.dims[3]);
+
+    try {
+        getQueue().submit([&](auto &h) {
+            sycl::accessor d_out{*out.data, h};
```
Comment (syurkevi) [inline src/backend/oneapi/kernel/tile.hpp:99]: These should be with sycl::access::mode semantics

File: src/backend/oneapi/tile.cpp [hunk 2]
```diff
@@ -25,8 +25,9 @@ Array<T> tile(const Array<T> &in, const af::dim4 &tileDims) {
 
     Array<T> out = createEmptyArray<T>(oDims);
 
-    ONEAPI_NOT_SUPPORTED("tile Not supported");
-    // kernel::tile<T>(out, in);
+    if constexpr(!(std::is_same_v<T, double> || std::is_same_v<T, cdouble>)) {
```
Comment (syurkevi) [inline src/backend/oneapi/tile.cpp:28]: Let's wait for the resolution of #3342 to determine the best approach here

File: src/backend/oneapi/kernel/tile.hpp [hunk 3]
```diff
@@ -0,0 +1,110 @@
...
+#include <common/kernel_cache.hpp>
+#include <debug_oneapi.hpp>
+
+#include <string>
+#include <vector>
+
+namespace oneapi {
+namespace kernel {
+
+constexpr sycl::specialization_id<bool> compilingAsDouble;
```
Comment (syurkevi) [inline src/backend/oneapi/kernel/tile.hpp:23]: leftover from debug? don't see this used

File: src/backend/oneapi/tile.cpp (hunk 2 above)
Comment (pv-pterab-s) [reply src/backend/oneapi/tile.cpp:28]: umar confirmed this approach. without it, arrayfire outright crashes

File: src/backend/oneapi/kernel/tile.hpp (hunk 3 above)
Comment (pv-pterab-s) [reply src/backend/oneapi/kernel/tile.hpp]: good catch

File: src/backend/oneapi/kernel/tile.hpp (hunk 1 above)
Comment (pv-pterab-s) [reply src/backend/oneapi/kernel/tile.hpp]: done

**Comment by pv-pterab-s** [comment]
dropping

**Pull request closed without merging**